software/CONFIGURATION.md

```diff
@@ -53,3 +53,53 @@ shows the default configuration:
 ```
 
 - `volts_per_octave` must be one of `1.0` (Eurorack standard) or `1.2` (Buchla standard)
+
+
+# Accessing config members in Python code
+
+The firmware converts the JSON file into a `ConfigSettings` object, where the JSON keys are converted
+to Python attributes.  The attribute names are the same as their JSON strings, but converted to upper case (so as to
+appear as constants in the code) and with any alphanumeric characters replaced with `_` characters.  If the key starts
+with a number, a `K_` prefix is added. e.g.:
+
+- `language` -> `.LANGUAGE`
+- `display_channel` -> `.DISPLAY_CHANNEL`
+- `2pi` -> `K_2PI`
+- `max-frequency` -> `MAX_FREQUENCY`
+
+The `europi` namespace contains `.europi_config` and `.experimental_config` members that contain all of the
+configuration attributes described in the sections above:
+
+```python
+>>> from europi import europi_config
+>>> dir(europi_config)
+[
+  '__class__',
+  '__init__',
+  '__module__',
+  '__qualname__',
+  '__dict__',
+  'to_attr_name',
+  'CPU_FREQ',
+  'DISPLAY_CHANNEL',
+  'DISPLAY_HEIGHT',
+  'DISPLAY_SCL',
+  'DISPLAY_SDA',
+  'DISPLAY_WIDTH',
+  'EUROPI_MODEL',
+  'GATE_VOLTAGE',
+  'MAX_INPUT_VOLTAGE',
+  'MAX_OUTPUT_VOLTAGE',
+  'PICO_MODEL',
+  'ROTATE_DISPLAY'
+]
+```
+
+In code:
+
+```python
+import europi
```

Comment:
Probably, but I opted for this version to make it obvious what namespace `europi_config` is inside of. Anyone familiar with Python could see this and translate it into your version if they preferred.

Reply:
I guess that makes sense, it's just that everywhere else in this PR you do it the other way, and I'd expect new usages to do it that way as well. So it makes sense to me to have our documentation be consistent with our actual use cases.

Reply:
Changed it to use `from europi import europi_config` instead. That makes it equally clear what namespace `europi_config` is defined in, but keeps the code a little simpler. I just wanted to avoid the wildcard import since that's less insightful as an example.

Reply:
Yeah, agree. Wildcard imports are problematic. I avoid them whenever possible.